Templates/ObjC/Base/SharedObjC.template.tt

```diff
@@ -203,15 +208,15 @@ private string GetHydratedIVarFromDictionary(OdcmProperty prop)
 
     if ([{0} isKindOfClass:[NSArray class]]){{
         for (id {2} in {0}){{
-            [{0}Result addObject:{3}];
+            [{0}Result addObject:{2}];
         }}
     }}
 
     _{0} = {0}Result;
         ",
         GetObjCProperty(prop),
         prop.Name,
-        prop.Projection.Type.Name.ToLowerFirstChar(),
+        "temp"+prop.Projection.Type.Name.ToUpperFirstChar(),
```

Comment:
What does the "temp" prefix do for us?

Reply:
This code generated the getter methods for properties which are Arrays. 
A "for-each" loop iterates through the array to return all the items in these getters. 
In some of the models the individual item in array and the array itself were getting the same names. So to remove this conflict I have added a "temp" prefix. This is used inside our models to return the actual array with proper name and these temp variable do not get exposed.

Reply:
this is a nitpick, but perhaps we should use '_' per https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingIvarsAndTypes.html

Reply:
Variables prefixed with '_' denote the instance variables. In our case the variable is local to scope of if-block. 
This piece of code -
 `if([licenseDetails isKindOfClass:[NSArray class]]){
        for (id licenseDetails in licenseDetails){
            [licenseDetailsResult addObject:licenseDetails];
        }
    }` 

gets changed to this 
` if ([licenseDetails isKindOfClass:[NSArray class]]){
        for (id tempLicenseDetails in licenseDetails){
            [licenseDetailsResult addObject:tempLicenseDetails];
        }
    }`

If 'temp' prefix is not suitable, I can use 'single' or 'individual' also as prefix